gitlab/client.py

```diff
@@ -881,18 +881,16 @@ def http_list(
 
         page = kwargs.get("page")
 
-        if iterator and page is not None:
-            arg_used_message = f"iterator={iterator}"
-            utils.warn(
-                message=(
-                    f"`{arg_used_message}` and `page={page}` were both specified. "
-                    f"`{arg_used_message}` will be ignored and a `list` will be "
-                    f"returned."
-                ),
-                category=UserWarning,
-            )
+        if iterator:
+            if page is not None:
+                utils.warn(
+                    message=(
+                        f"`{iterator=}` and `{page=}` were both specified. "
+                        f"`{page=}` will be ignored."
+                    ),
+                    category=UserWarning,
+                )
```

Comment:
I'm ok with flipping the ignore logic here. with `iterator` we're a bit more explicit than previously with `as_list=False` which was in our old code, so IMO no issue if providing `iterator=True` really always returns an iterator.

Any concerns @JohnVillalovos? There's some more duplication in custom listing methods but we could try to remove some of that later via some kind of `self._get_list_path` helper perhaps - seems to be the only reason we override those for the most part.